### PR TITLE
Added a validation on the cost types

### PR DESCRIPTION
**Pull request by micheles**
The business_interruption cost_type gave in error in Anirudh's tests (https://ci.openquake.org/job/zdevel_oq-risk-tests/73/console) since it is 21 characters long and the limit was set to 20, so it was silently truncated.

### CODE REVIEW

File: openquake/commonlib/readinput.py [hunk 1]
```diff
@@ -670,12 +672,20 @@ def get_exposure_lazy(fname, ok_cost_types):
...
     cost_types = [(ct['name'], ct['type'], ct['unit'])
                   for ct in conversions.costTypes
                   if ct['name'] in ok_cost_types]
     if 'occupants' in ok_cost_types:
         cost_types.append(('occupants', 'per_area', 'people'))
     cost_types.sort(key=operator.itemgetter(0))
+    for row in cost_types:
+        valid.simple_id(row[0])  # the name
+        for col in row:
+            if len(col) > COST_TYPE_SIZE:
```
Comment (raoanirudh) [inline openquake/commonlib/readinput.py]: Presuming that we have already checked for the validity of the strings in cost_types, would we ever reach this point?

Comment (micheles) [inline openquake/commonlib/readinput.py]: You are right that the test should be done earlier. Let me see if I can move this up in the validation chain (right now there is no check at all).

**Comment by micheles** [comment]
Move the check earlier; now only structural,nonstructural,contents and business_interruption are accepted as cost types.

**Comment by raoanirudh** [comment]
Are occupants handled separately?

**Comment by micheles** [comment]
I am not happy about how occupants are managed (they should be more like cost types) but let leave them out of this pull request.

**Comment by raoanirudh** [comment]
LGTM

**Pull request merged**